packages/nuxt/src/app/composables/once.ts

```diff
@@ -24,9 +24,9 @@ export async function callOnce (...args: any): Promise<void> {
   if (nuxtApp.payload.once.has(_key)) {
     return
   }
+  nuxtApp.payload.once.add(_key)
   nuxtApp._once = nuxtApp._once || {}
   nuxtApp._once[_key] = nuxtApp._once[_key] || fn()
```

Comment:
How about instead:

```suggestion
  nuxtApp._once[_key] = nuxtApp._once[_key] || fn() || true
```

This adds a 'true' default value for non-async once callbacks, instead of early returning when a promise has not been resolved.

Otherwise we will not await the promise if one is running in parallel.

Reply:
Now I get it. When running in parallel all `callOnce` with the same key should resolve at the same time. Thanks for the clarification.